storage/file_persister.go

```diff
@@ -12,3 +15,49 @@ import (
 type FilePersister interface {
 	Persist(ctx context.Context, logger *log.Logger, path string, data io.Reader) error
 }
+
+// LocalFilePersister will persist files to the local disk.
+type LocalFilePersister struct{}
+
+// Persist will write the contents of data to the local disk on the specified path.
+// TODO: we should not write to disk here but put it on some queue for async disk writes.
+func (l *LocalFilePersister) Persist(_ context.Context, logger *log.Logger, path string, data io.Reader) error {
+	cp := filepath.Clean(path)
+
+	dir := filepath.Dir(cp)
+	if err := os.MkdirAll(dir, 0o755); err != nil {
+		return fmt.Errorf("creating a local directory %q: %w", dir, err)
+	}
+
+	f, err := os.OpenFile(cp, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0o600)
+	if err != nil {
+		return fmt.Errorf("creating a local file %q: %w", cp, err)
+	}
+	defer func() {
+		if err := f.Close(); err != nil {
+			logger.Errorf("LocalFilePersister:Persist", "closing the local file %q: %v", cp, err)
```

Comment:
I think we can handle this also from the call site if we return an error here. So we don't have to pass a logger here. The call site can log the error if it sees it's a file close error. WDYT?

Reply:
Yeah, that's a nice idea! Added it to cfa2726d577b1a26a0233ba5a10dd540ded70ffe

In which case the `logger` can be removed 🎉